src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformDependenciesHelper.kt

```diff
@@ -812,65 +899,147 @@ class IntelliJPlatformDependenciesHelper(
         )
     }
 
-    /**
-     * Collects all dependencies on plugins or modules of the current [IdePlugin].
-     * The [path] parameter is a list of already traversed entities, used to avoid circular dependencies when walking recursively.
-     *
-     * @param path IDs of already traversed plugins or modules.
-     */
-    private fun IdePlugin.collectDependencies(path: List<String> = emptyList()): List<IvyModule.Dependency> {
-        val id = requireNotNull(pluginId)
-        val dependencyIds = (dependencies.map { it.id } + optionalDescriptors.map { it.dependency.id } + modulesDescriptors.map { it.name } - id).toSet()
-        val buildNumber by lazy { productInfo.get().buildNumber }
-        val platformPath by lazy { platformPath.get() }
-
-        val plugins = dependencyIds
-            .mapNotNull { bundledPlugins[it] }
-            .map { plugin ->
-                val artifactPath = requireNotNull(plugin.originalFile)
-                val group = Dependencies.BUNDLED_PLUGIN_GROUP
-                val name = requireNotNull(plugin.pluginId)
-                val version = requireNotNull(plugin.pluginVersion)
-
-                writeIvyModule(group, name, version) {
-                    IvyModule(
-                        info = IvyModule.Info(group, name, version),
-                        publications = listOf(artifactPath.toIvyArtifact()),
-                        dependencies = when {
-                            id in path -> emptyList()
-                            else -> plugin.collectDependencies(path + id)
-                        },
-                    )
+    companion object {
+        private val log = Logger(IntelliJPlatformDependenciesHelper::class.java)
+
+        internal fun Path.getBundledPlugins(pluginManager: IdePluginManager): Map<String, IdePlugin> {
+            return this.resolve("plugins")
+                .listDirectoryEntries()
+                .filter { it.isDirectory() }
+                .mapNotNull { path ->
+                    // TODO: try not to parse all plugins at once
```

Comment:
Now it parses them only if `local("/opt/ideaIU")` is used. Otherwise it is done in the transformer only once.